lib/account/account-manager.js

```diff
@@ -46,24 +104,28 @@
  * @property {string} regId Unhashed registration id sent to client
  */
 
-module.exports = function (accountService, config, hotp, otDate, promise, random) {
+module.exports = function (accountService, base64, config, hotp, otDate, promise, random, secureHash) {
+    var accountConfig;
+
+    accountConfig = config.account || {};
+
     /**
      * Checks to see if the values for MFA are valid and will
      * throw if they are not.
      *
      * @param {string} mfaKey
-     * @param {string} previousMfa
      * @param {string} currentMfa
+     * @param {string} [previousMfa]
```

Comment:
How does JSON validation affect the usage of this function?  How do we enforce that the previous is required in places in our code?

I guess what I am getting at is that the manager should be the one that knows for a fact that a previous one is required or that a previous one is not required.  If split into two functions, checkMfa and checkMfaPrevious or something, then we can guarantee that it works.

I brought up this point because I could pass anything falsy - "" for instance, or null, and it will pass the validation function here but perhaps we are supposed to require the previous MFA.  If the JSON schema said that it had to be a string and we screwed around with it and changed it to a number, then I could bypass the previous MFA check by using "000000".